### PR TITLE
feat(docker): 3-stage uv build replacing vendor base image (WU-12)

### PR DESCRIPTION
**Pull request by MacAttak**
## Summary

- Replace `dagster/dagster-celery-k8s` vendor base image with 3-stage uv build from `python:3.11-slim`
- Add `[docker]` extras to orchestrator plugin for webserver/daemon/k8s
- Implement manifest-driven plugin selection in Makefile with multi-arch support
- 118 structural tests validate the new Dockerfile, supply chain security, and demo product layout

## Acceptance Criteria

| AC | Description | Status | Evidence |
|----|-------------|--------|----------|
| AC-12.1 | 3-stage Dockerfile (export, build, runtime) | PASS | `docker/dagster-demo/Dockerfile` — 3 named `FROM...AS` stages |
| AC-12.2 | uv export from workspace lockfile | PASS | Export stage uses `uv export --frozen --no-dev --package` |
| AC-12.3 | Docker extras for Dagster deps | PASS | `pyproject.toml` `[docker]` group with webserver/daemon/k8s |
| AC-12.4 | Manifest-driven plugin selection | PASS | Makefile reads `demo/manifest.yaml`, passes `FLOE_PLUGINS` build arg |
| AC-12.5 | Supply chain security | PASS | `--require-hashes`, `@sha256:` digest pins, `pip check` |
| AC-12.6 | Multi-arch build support | PASS | `--platform linux/amd64,linux/arm64` in Makefile |
| AC-12.7 | Selective source COPY (P16) | PASS | Only target package source copied in build stage |
| AC-12.8 | Demo product layout preserved | PASS | All 3 products, `__init__.py`, manifest, macros |
| AC-12.9 | Structural tests updated | PASS | 118 tests pass — new export/build/runtime/supply-chain tests |
| AC-12.10 | uv.lock updated | PASS | `dagster-webserver`, `dagster-daemon`, `dagster-k8s` in lockfile |

## Boundary Conditions

| BC | Description | Status |
|----|-------------|--------|
| BC-12.1 | Makefile chain preserved | PASS — `compile-demo → build-demo-image → demo` intact |
| BC-12.2 | Helm values unchanged | PASS — no changes to values-demo/test |
| BC-12.3 | Generated definitions unchanged | PASS — definitions tests pass unchanged |
| BC-12.4 | No vendor base image | PASS — no `FROM dagster/` in Dockerfile |

## Gate Results

| Gate | Status | Findings (B/W/I) |
|------|--------|-------------------|
| gate-build | PASS | 0/0/0 |
| gate-tests | WARN | 0/3/3 |
| gate-security | WARN | 0/2/3 |
| gate-wiring | WARN | 0/3/2 |
| gate-spec | PASS | 0/0/1 |

**0 BLOCK findings.** Remaining WARNs are pre-existing (WU-11) or out-of-scope (Helm default values).
Assertion strength warnings resolved in post-verify commit.

## Architecture Decisions

- **AD-1**: No vendor base image — build from `python:3.11-slim` for full control
- **AD-2**: `uv export` not `uv sync` — selective package installation from workspace lockfile
- **AD-3**: Docker extras pattern — runtime deps in `[docker]` optional group
- **AD-4**: 3-stage design — cache-optimized layers, minimal runtime attack surface

## Commits

1. `feat(orchestrator): add docker extras for webserver, daemon, and k8s` (T66)
2. `feat(docker): replace vendor base image with 3-stage uv build` (T67)
3. `feat(make): manifest-driven plugin selection for Docker build` (T68)
4. `test(packaging): update structural tests for 3-stage uv Dockerfile` (T69)
5. `chore: update secrets baseline for uv.lock changes` (T70)
6. `fix(packaging): strengthen assertions and resolve verify warnings`
7. `fix(packaging): remove dead tomli fallback (Python 3.11+ has tomllib)`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docker/dagster-demo/Dockerfile [hunk 1]
```diff
@@ -1,47 +1,137 @@
...
+# --require-hashes: verify every package against SHA256 hashes from uv export
+COPY --from=export /build/requirements.txt .
+RUN pip install --no-cache-dir --require-hashes -r requirements.txt
+
+# Copy workspace package source trees (selective per P16 — only installed packages)
 COPY packages/floe-core /build/packages/floe-core
+COPY plugins/floe-catalog-polaris /build/plugins/floe-catalog-polaris
 COPY plugins/floe-orchestrator-dagster /build/plugins/floe-orchestrator-dagster
 COPY plugins/floe-compute-duckdb /build/plugins/floe-compute-duckdb
 COPY plugins/floe-dbt-core /build/plugins/floe-dbt-core
```
[the remark this answers is not in the document]
Comment (MacAttak) [reply docker/dagster-demo/Dockerfile]: Addressed in 793617a:

1. **Documentation**: Added a comment block above the COPY list explaining the coupling:
   ```
   # NOTE: This list must include every package that may appear in FLOE_PLUGINS.
   # If a new plugin is added to demo/manifest.yaml, add its COPY line here too.
   # A missing COPY will cause the build to fail with "ERROR: source for $pkg not found".
   ```

2. **`floe-catalog-polaris`** is intentionally included in the COPY list but not in the default `FLOE_PLUGINS` ARG — it's pulled in when the Makefile overrides the ARG from `demo/manifest.yaml`. The inverse case (plugin in FLOE_PLUGINS but missing from COPY) produces a clear build-time error at line 106, which is the intended feedback mechanism.

Making the COPY list fully data-driven would require either a templating step or `COPY . /build/` (which violates P16 selective copy). The explicit list + build failure is the simpler trade-off for a demo image with a stable plugin set.
